// @framer/nvh.isometric/code/Isometric.tsx
import * as React from "react"
import {
    Frame,
    FramerEvent,
    PropertyControls,
    ControlType,
    addPropertyControls,
} from "framer"

interface Props {
    rotationX: number
    rotationY: number
    rotationZ: number
    children?: React.ReactNode
}

export function Isometric(props: Props) {
    const { rotationX, rotationY, rotationZ } = props
    return (
        <Frame
            width={"100%"}
            height={"100%"}
            rotateX={rotationX}
            rotateY={rotationY}
            rotateZ={rotationZ}
            scale={0.6}
            background={null}
        >
            {props.children}
        </Frame>
    )
}

Isometric.defaultProps = {
    rotationX: 50,
    rotationY: 0,
    rotationZ: 320,
}

addPropertyControls(Isometric, {
    rotationX: {
        type: ControlType.Number,
        title: "Rotate X",
        min: 0,
        max: 360,
    },
    rotationY: {
        type: ControlType.Number,
        title: "Rotate Y",
        min: 0,
        max: 360,
    },
    rotationZ: {
        type: ControlType.Number,
        title: "Rotate Z",
        min: 0,
        max: 360,
    },
})
